src/common/transformations/src/transformations/common_optimizations/activations_scaling.cpp

```diff
@@ -0,0 +1,434 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/activations_scaling.hpp"
+
+#include <memory>
+
+#include "itt.hpp"
+#include "low_precision/network_helper.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/convolution.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/group_normalization.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/mvn.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/unsqueeze.hpp"
+#include "openvino/op/variadic_split.hpp"
+#include "openvino/pass/constant_folding.hpp"
+#include "openvino/pass/manager.hpp"
+#include "openvino/pass/pattern/op/optional.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "ov_ops/rms.hpp"
+#include "transformations/common_optimizations/lin_op_sequence_fusion.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace {
+const auto is_scalar_node = [](const ov::Output<ov::Node>& output) -> bool {
+    const auto shape = output.get_partial_shape();
+    if (shape.is_dynamic())
+        return false;
+    return ov::shape_size(shape.to_shape()) == 1;
+};
+
+const auto is_non_const_node = [](const ov::Output<ov::Node>& output) -> bool {
+    return !ov::is_type<ov::op::v0::Constant>(output.get_node());
+};
+}  // namespace
+
+using namespace ov::pass::activations_scaling;
+using namespace ov::pass::pattern;
+using ov::pass::pattern::op::Or;
+
+// Add scale_down and scale_up layers around Convolution and MatMul nodes
+// Conv/MatMul
+//    ==>
+// Multiply(scale_down by scale_factor) --> Conv/MatMul --> Multiply(scale_up by scale_factor)
```

Comment:
Could you please move all the transformation descriptions to hpp file? I believe this helps easier to understand what each transformation does, since all the descriptions will suit on 1-2 screens

Reply:
I moved all descriptions to hpp file. Thank you.